### ISSUE TITLE
expiring_todo only works if it is the first TODO in the file

### ISSUE THREAD

**Issue by BarryBryant** [realm/SwiftLint#3767]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

The expiring_todo rule does not trigger when there are `// TODO:` comments above it in the file

##### Complete output when running SwiftLint, including the stack trace and command used
With a todo comment on a line above the todo with date, no expiring_todo violation
```bash
$ echo "// TODO: XYZ \n public enum Redacted { // TODO: [10/14/2019]" | Pods/SwiftLint/swiftlint line --no-cache --use-stdin --enable-all-rules

<nopath>:2:1: warning: Indentation Width Violation: Code should be indented using one tab or 4 spaces. (indentation_width)
<nopath>:2:9: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
<nopath>:1:4: warning: Todo Violation: TODOs should be resolved (XYZ). (todo)
<nopath>:2:28: warning: Todo Violation: TODOs should be resolved ([10/14/2019]). (todo)
```

With a normal comment on a line above the todo with date, correctly returns expiring_todo violation
```bash
$ echo "// XYZ \n public enum Redacted { // TODO: [10/14/2019]" | Pods/SwiftLint/swiftlint line --no-cache --use-stdin --enable-all-rules

<nopath>:2:35: error: ExpiringTodo Violation: TODO/FIXME has expired and must be resolved. (expiring_todo)
<nopath>:2:1: warning: Indentation Width Violation: Code should be indented using one tab or 4 spaces. (indentation_width)
<nopath>:2:9: warning: Missing Docs Violation: public declarations should be documented. (missing_docs)
<nopath>:2:28: warning: Todo Violation: TODOs should be resolved ([10/14/2019]). (todo)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.45.0
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods

**Comment by serges147** [comment]
Finally, I understand why some of my `// TODO: [mm/dd/yyyy] Xxx` are not triggered while others work perfectly! Exactly as described above, all you need is just at least two such todos in a swift file.

**Comment by serges147** [comment]
I found pretty simple solution, preparing PR...

**Comment by serges147** [comment]
https://github.com/realm/SwiftLint/pull/3803
